.openapidoc/index.js

```diff
@@ -0,0 +1,142 @@
+const fs = require("fs");
+const yaml = require("js-yaml");
+const fetch = require("node-fetch");
+const ghpages = require("gh-pages");
+
+const config = new (function () {
+  this.repo = "git@github.com:PegaSysEng/eth2signer.git";
+  this.branch = "gh-pages";
+  this.specPath = "../core/build/resources/main/openapi/eth2signer.yaml";
+  this.specVersion = "";
+  this.versionsFileName = "versions.json";
+  this.versionsFileUrl = `https://github.com/PegaSysEng/eth2signer/raw/${this.branch}/${this.versionsFileName}`;
+  this.distDir = "./dist";
+  this.versionsFileDist = `${this.distDir}/${this.versionsFileName}`;
+})();
+
+/**
+ * Main function to prepare and publish openapi spec to gh-pages branch
+ */
+async function main() {
+  prepareDistDir();
+  calculateVersionFromSpec();
+  copySpecFileToDist();
+  await updateVersionJson();
+  cleanGhPagesCache();
+  await publishToGHPages();
+}
+
+/**
+ * Re-create dist dir
+ */
+function prepareDistDir() {
+  fs.rmdirSync(config.distDir, { recursive: true });
+  fs.mkdirSync(config.distDir, { recursive: true });
+}
+
+/**
+ * Parse OpenApi yaml and update config spec version.
+ */
+function calculateVersionFromSpec() {
+  const data = yaml.safeLoad(fs.readFileSync(config.specPath, "utf8"));
+  config.specVersion = data.info.version;
+}
+
+/**
+ * Determine if its a release version from spec version. It contains -dev- as a result from our gradle build
+ */
+function isReleaseVersion() {
+  return !config.specVersion.includes("-dev-");
```

Comment:
Don't you base your releases on tags? A doc for a released version should not change once published. Here it means that you will have the doc tied to updates on the branch and not only on the final tagging action. It may be better, I don't know, but it must be clearly expected.